lib/hutch/broker.rb

```diff
@@ -382,5 +384,32 @@ def consumer_pool_size
     def consumer_pool_abort_on_exception
       @config[:consumer_pool_abort_on_exception]
     end
+
+    def cancel_all_channel_consumers
+      return unless channel&.consumers
+
+      logger.debug("Broker: Cancel channel.consumers: #{channel.consumers.inspect}")
+
+      channel.consumers.each_value { |consumer| cancel(consumer) }
+    end
+
+    #
+    # Cancel a consumer with retry.
+    #
+    # @param [Bunny::Consumer] consumer
+    #
+    def cancel(consumer)
```

Comment:
This entire function is not necessary, it does not do anything useful over `Bunny::Consumer#cancel`, the retries are infinite and can prevent the process from shutting down.

Retrying consumer cancellation right before all channels and connections shut down does not make much sense, regardless of what Sneakers might do (Pika also cancels consumers on shutdown and it resulted a unique, Pika-speicific issues in certain cases).

Finally, the consumer cannot be `nil`, you are calling this method while iterating on a collection that supposedly cannot be accessed concurrently.